x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportDeleteExpiredDataAction.java

```diff
@@ -164,7 +172,25 @@ void deleteExpiredData(Iterator<MlDataRemover> mlDataRemoversIterator,
             if (haveAllPreviousDeletionsCompleted) {
                 logger.info("Completed deletion of expired ML data");
             } else {
-                logger.info("Halted deletion of expired ML data until next invocation");
+                if (isTimedOutSupplier.get()) {
+                    String msg = "Deleting expired ML data was cancelled after the timeout period of [" +
+                        MlDataRemover.DEFAULT_MAX_DURATION  + "] was exceeded. " +
+                        "The setting [xpack.ml.nightly_maintenance_requests_per_second] " +
+                        "controls the deletion rate, consider increasing the value to assist in pruning old data";
+                    logger.warn(msg);
+
+                    if (Strings.isNullOrEmpty(request.getJobId())
+                        || Strings.isAllOrWildcard(new String[]{request.getJobId()})
+                        || request.getExpandedJobIds() == null) {
+                        auditor.warning(AbstractAuditor.All_RESOURCES_ID, msg);
+                    } else {
+                        for (String jobId : request.getExpandedJobIds()) {
+                            auditor.warning(jobId, msg);
+                        }
+                    }
+                } else {
+                    logger.info("Halted deletion of expired ML data until next invocation");
```

Comment:
What would cause it to halt before completed for anything other than an expiration?

Reply:
Nothing, at the moment the data deleters will only return false on timeout. But that may not be the case forever and I didn't want to change the existing log messages too much